Merge

h9k opened the pull request:
Merged all extra functionality I had in my own module.
Each method has its own commit for easier review, later we can rebase.

papakai reviewed (changes requested):
a) as discussed: rewording the commit messages to our used format
b) I am ok with how it is right now, but fyi: at HEXONET we have possibilities to check if AuthCode is required for a Transfer, if the provided authcode is correct etc. by using `CheckDomainTransfer` command. Maybe this also exists for RRP somehow.
c) like b), checking if domain update requires a trade, can be done at HEXONET using an API Command. As far as I remember it is the API Command `QueryDomainOptions`. Probably RRP provides something similar

b) and c) are just remarks, nothing I ask you to dive into. Just things that are eventually dropping out manual effort to update such static configurations.

h9k commented:
> a) as discussed: rewording the commit messages to our used format
> b) I am ok with how it is right now, but fyi: at HEXONET we have possibilities to check if AuthCode is required for a Transfer, if the provided authcode is correct etc. by using `CheckDomainTransfer` command. Maybe this also exists for RRP somehow.
> c) like b), checking if domain update requires a trade, can be done at HEXONET using an API Command. As far as I remember it is the API Command `QueryDomainOptions`. Probably RRP provides something similar
> 
> b) and c) are just remarks, nothing I ask you to dive into. Just things that are eventually dropping out manual effort to update such static configurations.
> 
> Feel free to merge after a) is covered.

I have updated the commit messages and renamed the module as discussed.
I will keep this PR as draft until I finished checking a few TODOs and tested the whole functionality.

papakai commented:
:tada: This PR is included in version 0.2.0 :tada:

The release is available on:
- [GitHub release](https://github.com/rrpproxy/whmcs-rrpproxy-registrar/releases/tag/v0.2.0)
- [WHMCS Marketplace Product Version](https://marketplace.whmcs.com/product/5573)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
